### PR TITLE
[Notifications] Add default values and validation to notifications

### PR DESCRIPTION
**Pull request by yaelgen**
resolves: [ML-4090](https://jira.iguazeng.com/browse/ML-4090) and [ML-4091](https://jira.iguazeng.com/browse/ML-4091)

The following fields must be present in the notification: kind, when, message, and severity; the rest are optional.
This PR ensures that if they are not passed, the request will not fail and will have default values, and adds validation that it is as intended when they are.

Please note that this change affects backward compatibility, but the notification feature has not been released to users yet, so we can disregard the breakage.

### CODE REVIEW

File: tests/utils/test_notifications.py [hunk 1]
```diff
@@ -496,6 +544,25 @@ def test_notification_validation_on_object(
...
+        "condition": "",
+        "name": "",
+    }
+
+    for field, expected_value in notification_fields.items():
+        value = getattr(notification, field)
+        assert value is not None, f"{field} field in the notification object is None"
+        assert (
+            value == expected_value
+        ), f"{field} field value is {value}, expected {expected_value}"
```
Comment (quaark) [inline tests/utils/test_notifications.py:563]: The first assert is a bit redundant when you have the second one

**Review by quaark — CHANGES_REQUESTED** [review]
Nice! minor thing

File: tests/utils/test_notifications.py (hunk 1 above)
Comment (yaelgen) [reply tests/utils/test_notifications.py:563]: what if the field is not None but is not the expected?

Comment (quaark) [reply tests/utils/test_notifications.py:563]: Your test isn't checking that scenario. It's testing an empty notification

File: mlrun/common/schemas/notification.py [hunk 2]
```diff
@@ -42,13 +42,13 @@ class NotificationStatus(mlrun.common.types.StrEnum):
...
-    when: typing.List[str] = None
-    condition: str = None
-    params: typing.Dict[str, typing.Any] = None
+    kind: NotificationKind
+    name: str
+    message: str
+    severity: NotificationSeverity
+    when: typing.List[str]
+    condition: str
+    params: typing.Dict[str, typing.Any] = {}
```
Comment (liranbg) [inline mlrun/common/schemas/notification.py:51]: that is a mutable default parameter.
In python, the mutable are computed once and each change is defined to all consecutive calls. default to None and enrich if is None.

**Review by liranbg — CHANGES_REQUESTED** [review]

**Review by quaark — APPROVED** [review]

**Pull request merged**